src/Simulation/Simulators.Tests/TestProjects/HoneywellExe/ClassicallyControlledSupportTests.qs

```diff
@@ -0,0 +1,732 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+
+namespace Microsoft.Quantum.Simulation.Testing.Honeywell.ClassicallyControlledSupportTests {
+
+    operation SubOp1() : Unit { }
+    operation SubOp2() : Unit { }
+    operation SubOp3() : Unit { }
+
+    operation SubOpCA1() : Unit is Ctl + Adj { }
+    operation SubOpCA2() : Unit is Ctl + Adj { }
+    operation SubOpCA3() : Unit is Ctl + Adj { }
+    
+    operation BasicLift() : Unit {
+        let r = Zero;
+        if (r == Zero) {
+            SubOp1();
+            SubOp2();
+            SubOp3();
+            let temp = 4;
+            using (q = Qubit()) {
+                let temp2 = q;
+            }
+        }
+    }
+
+    operation LiftLoops() : Unit {
+        let r = Zero;
```

Comment:
Would you mind modifying some of these to just actually allocate and measure a qubit? Now that we actually tests execution, it seems like a good idea to check that all works as expected with allocations and qubit use as well.

Reply:
I can add the tests that IonQ has to Honeywell and QCI. Would that address this?

Reply:
Oh I see what you mean, instead of `let r = Zero;`, you want `let r = M(q);`, right?

Reply:
Maybe that should be tested separately anyway though? If I leave the tests the way they are and add the tests from IonQ, then, if something goes wrong with the measurement on a qubit, just the new tests would fail and it would be more informative. Or if something goes wrong with the conditionals, then just the relevant tests would fail and the tests from IonQ would still pass. I think this would be more informative have more atomicity in the tests. I will add the tests from IonQ to Honeywell and QCI.

Reply:
I've added the tests from IonQ that will check that measurement of qubits works.

Reply:
Adding the tests for IonQ sounds good, and adding a separate tests where you actually branch based on the measurement outcome for an allocated qubit would also be a good thing to do.

Reply:
Done